cmd/mdatagen/internal/metadata/generated_metrics.go

```diff
@@ -342,7 +294,6 @@ func WithStartTimeOverride(start pcommon.Timestamp) ResourceMetricsOption {
 func (mb *MetricsBuilder) EmitForResource(rmo ...ResourceMetricsOption) {
```

Comment:
@djaglowski yes, options for Emit functions will be removed in the next PR. You can see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24444. But, for now, we still have `WithResource` and `WithStartTimeOverride` options. I just want to reduce the amount of changes in each PR. Otherwise, it's hard to review